okhttp/src/main/java/okhttp3/internal/http/CallServerInterceptor.kt

```diff
@@ -0,0 +1,129 @@
+    val exchange = realChain.exchange()
+    val request = realChain.request()
+    val requestBody = request.body()
+    val sentRequestMillis = System.currentTimeMillis()
+
+    exchange.writeRequestHeaders(request)
+
+    var responseHeadersStarted = false
+    var responseBuilder: Response.Builder? = null
+    if (HttpMethod.permitsRequestBody(request.method()) && requestBody != null) {
+      // If there's a "Expect: 100-continue" header on the request, wait for a "HTTP/1.1 100
+      // Continue" response before transmitting the request body. If we don't get that, return
+      // what we did get (such as a 4xx response) without ever transmitting the request body.
+      if ("100-continue".equals(request.header("Expect"), ignoreCase = true)) {
+        exchange.flushRequest()
+        responseHeadersStarted = true
+        exchange.responseHeadersStart()
+        responseBuilder = exchange.readResponseHeaders(true)
+      }
+      if (responseBuilder == null) {
+        if (requestBody.isDuplex()) {
+          // Prepare a duplex body so that the application can send a request body later.
+          exchange.flushRequest()
+          val bufferedRequestBody = exchange.createRequestBody(request, true).buffer()
+          requestBody.writeTo(bufferedRequestBody)
+        } else {
+          // Write the request body if the "Expect: 100-continue" expectation was met.
+          val bufferedRequestBody = exchange.createRequestBody(request, false).buffer()
+          requestBody.writeTo(bufferedRequestBody)
+          bufferedRequestBody.close()
+        }
+      } else {
+        exchange.noRequestBody()
+        if (!exchange.connection().isMultiplexed) {
+          // If the "Expect: 100-continue" expectation wasn't met, prevent the HTTP/1 connection
+          // from being reused. Otherwise we're still obligated to transmit the request body to
+          // leave the connection in a consistent state.
+          exchange.noNewExchangesOnConnection()
+        }
+      }
+    } else {
+      exchange.noRequestBody()
+    }
+
+    if (requestBody == null || !requestBody.isDuplex()) {
+      exchange.finishRequest()
+    }
+    if (!responseHeadersStarted) {
+      exchange.responseHeadersStart()
+    }
+    if (responseBuilder == null) {
+      // exchange.readResponseHeaders(false) never returns null
+      // while exchange.readResponseHeaders(true) always returns null
+      responseBuilder = exchange.readResponseHeaders(false)
+    }
+      // safe to use!!
+    var response = responseBuilder!!
+        .request(request)
+        .handshake(exchange.connection().handshake())
+        .sentRequestAtMillis(sentRequestMillis)
+        .receivedResponseAtMillis(System.currentTimeMillis())
+        .build()
+    var code = response.code()
+    if (code == 100) {
+      // server sent a 100-continue even though we did not request one.
+      // try again to read the actual response
+      // exchange.readResponseHeaders(false) never be null
```

Comment:
Please remove this comment.

Reply:
I don't like the comment because it acts like a Jedi mind trick. “this value is not null” in a comment doesn't make it so.

Reply:
Fixed